### ISSUE TITLE
Will cause Dangling Pointer In member "uniques" of struct lysc_node_list when using devaition "not supported".

### ISSUE THREAD

**Issue by lyzliyuzhi91** [CESNET/libyang#2202]
https://github.com/CESNET/libyang/blob/fc4dbd923e044006c93df020590a1e5a8656c09e/src/schema_compile.c#L1376

When calling `lys_compile_node_list_unique`,  the unique node will be added to (struct lys_node_list *)->uniques array. 
But when I use a "not suppported" deviation, such node won't be removed from the array.

example yang:
```
module test-module {
  namespace "urn:test:md-omu:sdk:test-module";
  prefix test;
  list l1 {
    key "k1";
    unique "v1 v2";
    leaf k1 { type string; }
    leaf v1 { type string; }
    leaf v2 { type string; }
    leaf v3 { type string; }
  }
  deviation "/test:l1/test:v2" {
    deviate not-supported;
  }
}
```
If I access `v2` through `unique` statement, it will encounter dangling pointer.
Thanks.

**Comment by michalvasko** [comment]
Right, should be fixed but not sure if exactly the way you wanted. This YANG module will now fail to be loaded and a separate deviation deleting the `unique` statement is required, which seemed to me the correct behavior.

**Comment by lyzliyuzhi91** [comment]
ok
